src/devices_models/devices/hydro_generation.jl

```diff
@@ -11,164 +11,95 @@ struct HydroCommitmentRunOfRiver <: AbstractHydroUnitCommitment end
 struct HydroCommitmentReservoirBudget <: AbstractHydroUnitCommitment end
 struct HydroCommitmentReservoirStorage <: AbstractHydroUnitCommitment end
 
-########################### Hydro generation variables #################################
+########################### ActivePowerVariable, HydroGen #################################
 
-"""
-This function add the variables for active power to the model
-"""
-function AddVariableSpec(
-    ::Type{T},
-    ::Type{U},
-    ::PSIContainer,
-) where {T <: ActivePowerVariable, U <: PSY.HydroGen}
-    return AddVariableSpec(;
-        variable_name = make_variable_name(T, U),
-        binary = false,
-        expression_name = :nodal_balance_active,
-        initial_value_func = x -> PSY.get_active_power(x),
-        lb_value_func = x -> PSY.get_active_power_limits(x).min,
-        ub_value_func = x -> PSY.get_active_power_limits(x).max,
-    )
-end
+get_variable_binary(::ActivePowerVariable, ::Type{<:PSY.HydroGen}) = false
+get_variable_expression_name(::ActivePowerVariable, ::Type{<:PSY.HydroGen}) = :nodal_balance_active
 
-"""
-This function add the variables for reactive power to the model
-"""
-function AddVariableSpec(
-    ::Type{T},
-    ::Type{U},
-    ::PSIContainer,
-) where {T <: ReactivePowerVariable, U <: PSY.HydroGen}
-    return AddVariableSpec(;
-        variable_name = make_variable_name(T, U),
-        binary = false,
-        expression_name = :nodal_balance_reactive,
-        initial_value_func = x -> PSY.get_reactive_power(x),
-        lb_value_func = x -> PSY.get_reactive_power_limits(x).min,
-        ub_value_func = x -> PSY.get_reactive_power_limits(x).max,
-    )
-end
+get_variable_initial_value(pv::ActivePowerVariable, d::PSY.HydroGen, settings) =
+    get_variable_initial_value(pv, d, get_warm_start(settings) ? WarmStartVariable() : ColdStartVariable())
+get_variable_initial_value(::ActivePowerVariable, d::PSY.HydroGen, ::WarmStartVariable) = PSY.get_active_power(d)
+get_variable_initial_value(::ActivePowerVariable, d::PSY.HydroGen, ::ColdStartVariable) = nothing
 
-"""
-This function add the variables for energy storage to the model
-"""
-function AddVariableSpec(
-    ::Type{T},
-    ::Type{U},
-    ::PSIContainer,
-) where {T <: EnergyVariable, U <: PSY.HydroGen}
-    return AddVariableSpec(;
-        variable_name = make_variable_name(T, U),
-        binary = false,
-        initial_value_func = x -> PSY.get_initial_storage(x),
-        lb_value_func = x -> 0.0,
-        ub_value_func = x -> PSY.get_storage_capacity(x),
-    )
-end
+get_variable_lower_bound(::ActivePowerVariable, d::PSY.HydroGen, _) = PSY.get_active_power_limits(d).min
```

Comment:
Can you make another function here that dispatches on the formulation `HydroDispatchRunOfRiver` to return 0.0. This is related to #563.

Reply:
Resolved in https://github.com/NREL-SIIP/PowerSimulations.jl/pull/560/commits/28d99027f2bd2e4140806e0cdb92ebfed64f50cf.

Reply:
Can you clarify exactly what needs to be added here?

Reply:
add 

```julia
get_variable_lower_bound(::ActivePowerVariable, d::PSY.HydroGen, ::HydroDispatchRunOfRiver) = 0.0
```

Reply:
Resolved in https://github.com/NREL-SIIP/PowerSimulations.jl/pull/560/commits/7bf83ef6f1dc9396039e3358d0a44824f10690a5